### PR TITLE
chore(deps): migrate from yarn to npm

### PR DESCRIPTION
**Pull request by caugner**
### Description

1. Remove `yarn.lock`.
2. Run `npm install`.
3. Replace yarn commands in docs, scripts and workflows.

### Motivation

- In https://github.com/orgs/mdn/discussions/108, we decided to move to npm.
- bob has already been migrated in https://github.com/mdn/bob/pull/797.
- interactive-examples still needed to be migrated.

### CODE REVIEW

**Comment by NiedziolkaMichal** [comment]
I think it would be better to remove `, as it is in the local server that's started by yarn start` from `CSS-Example-Style-Guide.md`.

File: package.json [hunk 1]
```diff
@@ -15,14 +15,14 @@
     "main": "index.js",
     "scripts": {
         "build": "mdn-bob",
-        "fix:css": "yarn lint:css --fix",
+        "fix:css": "npm run lint:css --fix",
```
Comment (nschonni) [inline package.json]: Maybe it's changed, but I'm pretty sure you need the extra `--` to pass arguments to the tasks inside an NPM call
```suggestion
        "fix:css": "npm run lint:css -- --fix",
```

**Comment by caugner** [comment]
Triggered a stage deployment in this branch:

1. ❌ https://github.com/mdn/interactive-examples/actions/runs/4115698157
2. ❌ https://github.com/mdn/interactive-examples/actions/runs/4115715668
3. ✅ https://github.com/mdn/interactive-examples/actions/runs/4115753143

File: .github/workflows/stage-deploy.yml [hunk 2]
```diff
@@ -23,13 +23,14 @@ jobs:
...
         with:
           node-version: '16'
-          cache: yarn
+          cache: npm
 
-      - name: Install all yarn packages
-        run: yarn --frozen-lockfile
+      - name: Install all npm packages
+        run: npm ci --only=production
+        working-directory: bob
```
Comment (caugner) [inline .github/workflows/stage-deploy.yml]:
```suggestion
```

File: .github/workflows/stage-deploy.yml [hunk 3]
```diff
@@ -23,13 +23,13 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: '16'
-          cache: yarn
+          cache: npm
 
-      - name: Install all yarn packages
-        run: yarn --frozen-lockfile
+      - name: Install all npm packages
+        run: npm ci --only=production
```
Comment (caugner) [inline .github/workflows/stage-deploy.yml]:
```suggestion
        run: npm ci
```

**Comment by caugner** [comment]
@NiedziolkaMichal I have updated `CSS-Example-Style-Guide.md` and tested this PR using the stage-deploy workflow.

**Review by NiedziolkaMichal — APPROVED** [review]

**Review by nschonni — APPROVED** [review]
LGTM, looks like the unrelated build warnings are from the missing updated tags upstream for the AWS task https://github.com/aws-actions/configure-aws-credentials/issues/632

**Pull request merged**